src/libspark/chaum.cpp

```diff
@@ -43,7 +42,7 @@ void Chaum::prove(
         throw std::invalid_argument("Bad Chaum statement!");
     }
     for (std::size_t i = 0; i < n; i++) {
-        if (!(F*x[i] + G*y[i] + H*z[i] == S[i] && T[i]*x[i] + G*y[i] == U)) {
+        if (!(F*x[i] + G*y[i] + H*z[i] == S[i] && (T[i]*x[i] + G*y[i]).isInfinity())) {
```

Comment:
this condition forces if x[i] is zero then so is y[i] and vice versa. in that case the distribution of x and y might not be uniformly distributed.